modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APIFilterRelevantObjectEntryModelListenerTest.java

```diff
@@ -165,6 +165,68 @@ public void testPostFilter() throws Exception {
 				"headless-builder/filters", Http.Method.POST
 			).toString(),
 			JSONCompareMode.LENIENT);
+
+		JSONObject jsonObject = HTTPTestUtil.invokeToJSONObject(
+			JSONUtil.put(
+				"applicationStatus", "published"
+			).put(
+				"baseURL", StringUtil.toLowerCase(RandomTestUtil.randomString())
+			).put(
+				"externalReferenceCode", RandomTestUtil.randomString()
+			).put(
+				"title", RandomTestUtil.randomString()
+			).toString(),
+			"headless-builder/applications", Http.Method.POST);
+
+		JSONObject apiEndpointJSONObject = HTTPTestUtil.invokeToJSONObject(
+			JSONUtil.put(
+				"description", "description"
+			).put(
+				"externalReferenceCode", RandomTestUtil.randomString()
+			).put(
+				"externalReferenceCode", RandomTestUtil.randomString()
+			).put(
+				"httpMethod", "get"
+			).put(
+				"name", "name"
+			).put(
+				"path",
+				StringBundler.concat(
+					StringPool.FORWARD_SLASH,
+					RandomTestUtil.randomString(
+					).toLowerCase(),
+					"/{pathParameterId}")
+			).put(
+				"pathParameter", "id"
+			).put(
+				"r_apiApplicationToAPIEndpoints_c_apiApplicationId",
+				jsonObject.get("id")
+			).put(
+				"retrieveType", "singleElement"
+			).put(
+				"scope", APIApplication.Endpoint.Scope.COMPANY.getValue()
+			).toString(),
+			"headless-builder/endpoints", Http.Method.POST);
+
+		JSONAssert.assertEquals(
+			JSONUtil.put(
+				"status", "BAD_REQUEST"
+			).put(
+				"title",
+				"An API filter can only be associated to API endpoints with " +
```

Comment:
Sort the value of the error messages, because BC is always asking for doing that